packetbeat/beater/processor.go

```diff
@@ -92,6 +98,14 @@ func (p *processor) Stop() {
 		time.Sleep(p.shutdownTimeout)
 	}
 	p.publisher.Stop()
+	p.UpdateStatus(status.Stopped, "stopped packetbeat processor")
+}
+
+// UpdateStatus wraps the status reporter we get from central management
+func (p *processor) UpdateStatus(status status.Status, message string) {
+	if p.status != nil {
+		p.status.UpdateStatus(status, message)
+	}
```

Comment:
If you add a `noopReporter` type that has the implementation `func(noopReporter) UpdateStatus(status.Status, string) {}`, then on construction of the sniffer, if the provided `status.StatusReporter` is nil, it gets replaced with the noop. Then all the nil guards can go away.

Reply:
Ah, true, `newProcesssor()` isn't publicly exported, so I suppose that's fairly low-risk.